File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        count = {}
        res = 0
        l = 0

        for r in range(len(s)):
            if s[r] in count and count[s[r]] > 0:
                while s[r] in count and count[s[r]] > 0:
                    count[s[l]] -= 1
                    l += 1
            count[s[r]] = 1
            res = max(res, r - l + 1)

        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int res=0;
        Set<Integer> numSet = new HashSet<>();
        for(int n: nums){
            numSet.add(n);
        }
        for(int n : numSet){
            int len=1;

            if(!numSet.contains(n-1)){
                while(numSet.contains(n+len)){
                    len++;
                }
            }
            res=Math.max(res,len);
        }
        return res;
    }
}
